### PR TITLE
Remove obsolete comments in types.go

### LINKED ISSUE #4519: No defaulting in pkg/api/types.go means comments are lying
Many comments in there say "if not specified, default is ..." - but we previously asserted that there are no defaults for the internal API.  We should strip those comments out, and leave them in versioned structs only.

### PR DESCRIPTION
**Pull request by yujuhong**
This fixes #4519.

### CODE REVIEW

File: pkg/api/v1beta1/types.go [hunk 1]
```diff
@@ -61,7 +61,7 @@ type ContainerManifest struct {
 	Volumes       []Volume      `json:"volumes" description:"list of volumes that can be mounted by containers belonging to the pod"`
 	Containers    []Container   `json:"containers" description:"list of containers belonging to the pod"`
 	RestartPolicy RestartPolicy `json:"restartPolicy,omitempty" description:"restart policy for all containers within the pod; one of RestartPolicyAlways, RestartPolicyOnFailure, RestartPolicyNever"`
-	// Optional: Set DNS policy.  Defaults to "ClusterFirst"
+	// Required: Set DNS policy.
```
Comment (bgrant0607) [inline pkg/api/v1beta1/types.go]: By definition, defaulted fields are not required in the versioned types.go files.

Comment (yujuhong) [inline pkg/api/v1beta1/types.go]: I misunderstood the meaning of the comment. The defaults are applied when converting a versioned object to the internal one, so they are defaulted. I will upload a new patch.

**Comment by bgrant0607** [comment]
LGTM

**Comment by thockin** [comment]
If every field is required is it worth spelling out? :)

**Comment by bgrant0607** [comment]
Maybe not, though it is a change, so perhaps useful for a while.

**Comment by yujuhong** [comment]
@thockin, I couldn't find a consistent pattern of when to comment in the file. There are "required" and "optional" in multiple places, irregardless of this PR. Agreed that we probably should clean up all of them at once :)

By the way, there are still some optional fields with default values. For example, some bool fields are defaulted to false. I believe this depends on the implicit struct initialization behavior in golang. Is that true/intentional?

**Comment by thockin** [comment]
Yeah, there's not much we can do about zero-values short of running all
construction through functions with explicit arguments.

**Pull request merged**